internal/storage/rocksdb/store.go

```diff
@@ -508,6 +571,26 @@ func toByteArray(value *gorocksdb.Slice) []byte {
 	return res
 }
 
+const tsLength = 8
+const endOfWorld = 253402300799 // 9999-12-31 23:59:59 B{ 127 65 244 255 58 0 0 0 }
+const bitMarker = 200
+
+func getExpireTs(valueWithTtl []byte) (int64, error) {
+	l := len(valueWithTtl)
+	if l < tsLength {
+		return 0, fmt.Errorf("invalid_length")
+	}
+	ts := valueWithTtl[l-tsLength:]
+	if ts[7] == bitMarker {
```

Comment:
Though it is very rare for user data to fit this pattern, I think it might be useful to think about a header based approach. This header can contain info about how many bytes is the value and how many bytes is the expire_ts. We can have this header prefix for all kinds of values.